src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/ChatOptions.cs

```diff
@@ -9,6 +9,9 @@ namespace Microsoft.Extensions.AI;
 /// <summary>Represents the options for a chat request.</summary>
 public class ChatOptions
 {
+    /// <summary>Gets or sets an optional identifier used to associate a request with an existing conversation.</summary>
+    public string? ConversationId { get; set; }
```

Comment:
Per our recent offline conversation, this feels to me like information that could be encapsulated by a particular `IChatClient` implementation. I know that our approach so far has been to offload all context minus connection strings to the message log and options, but it makes me wonder if the new assistants API is signal that we might want to consider encapsulating information pertaining to a specific session/thread/conversation.